### PR TITLE
[COW-SDK] - replace SupportedChainId and change xDai > Gnosis Chain

### PR DESCRIPTION
**Pull request by W3stside**
# Summary

Part of the cow-sdk integration

- this replaces the SupportedChainId inside constants/chains with the cow-sdk one
- also changes any references of `xdai` to `gnosis_chain`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,6 +29,7 @@
     "@babel/preset-env": "^7.16.11",
     "@babel/preset-react": "^7.16.7",
     "@babel/preset-typescript": "^7.16.7",
+    "@cowprotocol/cow-sdk": "^0.0.9",
```
Comment (alfetopito) [inline package.json:32]: Shouldn't the minimum be 0.0.10 or 11 that you recently published?

File: src/custom/constants/chains/chainsMod.ts [hunk 2]
```diff
@@ -3,31 +3,14 @@
...
-
-  OPTIMISM = 10,
-  OPTIMISTIC_KOVAN = 69,
-
-  POLYGON = 137,
-  POLYGON_MUMBAI = 80001, */
-
-  XDAI = 100,
-}
+export { SupportedChainId } from '@cowprotocol/cow-sdk'
```
Comment (alfetopito) [inline src/custom/constants/chains/chainsMod.ts:9]: Since it's a mod, leave the code commented out rather than remove it

**Review by alfetopito — APPROVED** [review]
Working fine. Just 2 small comments

File: src/custom/api/gnosisSafe/index.ts [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
-  [ChainId.XDAI]: 'https://safe-transaction.xdai.gnosis.io',
+  [ChainId.GNOSIS_CHAIN]: 'https://safe-transaction.xdai.gnosis.io',
 }
 
 const SAFE_BASE_URL = 'https://gnosis-safe.io'
 const CHAIN_SHORT_NAME: Partial<Record<number, string>> = {
   [ChainId.MAINNET]: 'eth', // https://github.com/ethereum-lists/chains/blob/master/_data/chains/eip155-1.json
   [ChainId.RINKEBY]: 'rin', // https://github.com/ethereum-lists/chains/blob/master/_data/chains/eip155-4.json
-  [ChainId.XDAI]: 'xdai', // https://github.com/ethereum-lists/chains/blob/master/_data/chains/eip155-100.json
+  [ChainId.GNOSIS_CHAIN]: 'xdai', // https://github.com/ethereum-lists/chains/blob/master/_data/chains/eip155-100.json
```
Comment (nenadV91) [inline src/custom/api/gnosisSafe/index.ts:15]: Isn't the short name actually gno, based on this https://github.com/ethereum-lists/chains/blob/71bbadb6878db890f2763c8a84b5ed0b94d0e353/_data/chains/eip155-100.json#L21 ?

File: src/custom/api/gnosisProtocol/api.ts [hunk 4]
```diff
@@ -38,15 +38,15 @@ function getGnosisProtocolUrl(): Partial<Record<ChainId, string>> {
...
+      [ChainId.GNOSIS_CHAIN]: process.env.REACT_APP_API_URL_STAGING_XDAI || 'https://barn.api.cow.fi/xdai/api',
     }
   }
 
   // Production, staging, ens, ...
   return {
     [ChainId.MAINNET]: process.env.REACT_APP_API_URL_PROD_MAINNET || 'https://api.cow.fi/mainnet/api',
     [ChainId.RINKEBY]: process.env.REACT_APP_API_URL_PROD_RINKEBY || 'https://api.cow.fi/rinkeby/api',
-    [ChainId.XDAI]: process.env.REACT_APP_API_URL_PROD_XDAI || 'https://api.cow.fi/xdai/api',
+    [ChainId.GNOSIS_CHAIN]: process.env.REACT_APP_API_URL_PROD_XDAI || 'https://api.cow.fi/xdai/api',
```
Comment (nenadV91) [inline src/custom/api/gnosisProtocol/api.ts:49]: Should we maybe update the env variable names for consistency or is it fine to leave it as this?

**Review by nenadV91 — APPROVED** [review]
Approved with some minor comments

File: src/custom/api/gnosisSafe/index.ts (hunk 3 above)
Comment (alfetopito) [reply src/custom/api/gnosisSafe/index.ts:15]: It is, but this is only internal.
While I do agree with you and this is mildly annoying, I'm fine to have the long version

File: src/custom/api/gnosisProtocol/api.ts (hunk 4 above)
Comment (alfetopito) [reply src/custom/api/gnosisProtocol/api.ts:49]: Agreed, will create an issue to address it post merge

Comment (alfetopito) [reply src/custom/api/gnosisProtocol/api.ts:49]: https://github.com/cowprotocol/cowswap/issues/579

**Pull request merged**